packages/components/filters/filter-menu/src/filter-menu.tsx

```diff
@@ -0,0 +1,109 @@
+import { type ReactElement } from 'react';
+import DropdownMenu from '@commercetools-uikit/dropdown-menu';
+import { Footer } from './footer';
+import { Header } from './header';
+import { TriggerButton } from './trigger-button';
+
+export type TAppliedFilterValue = {
+  key?: string;
+  label: string | ReactElement;
+};
+
+export type TFilterMenuProps = {
+  /**
+   * unique identifier for the filter
+   */
+  filterKey: string;
+  /**
+   * formatted message to display the filter's name
+   */
+  label: string | ReactElement;
+  /**
+   * the input in which the user selects values for the filter
+   *
+   * NOTE / OPINIONS WELCOME IN PR COMMENTS:
+   * can either pass a react element and let the consuming app choose what input to use,
+   * or start defining different menu bodies based on props and only make predefined input
+   * types available, which is restrictive and much higher maintenance
+   */
+  filter: ReactElement;
+  /**
+   * the input in which the user can select which operator should be used for this filter
+   *
+   * NOTE / OPINIONS WELCOME IN PR COMMENTS:
+   * the alternative to a rect element is passing in an array of operators and a handler function,
+   * which is more achievable here than for the input in the body, but could still be restrictive
+   */
+  operatorsInput?: ReactElement;
```

Comment:
The inputs never really change, only their payload, right? If so I'd go with the array config.